### ISSUE TITLE
Why does my SwipeView page not change when I change the Tab in the TabBar in QML?

### ISSUE THREAD

**Issue by gravetii** [therecipe/qt#1005]
So, I'm trying to create a traditional look for my QML application which simply consists of a `TabBar` and a `SwipeView`. There's also another component which essentially boils down to a button click which should result in a new tab in the `TabBar` and a new page in the `SwipeView`. This works, except that when I click on the previous tab (there's a static tab and page when the app starts up), the page in the view doesn't change.

Here's my QML file - 

```
import QtQuick 2.12
import QtQuick.Layouts 1.3
import QtGraphicalEffects 1.12
import QtQuick.Controls 2.13

ColumnLayout {
    Layout.fillHeight: true
    Layout.fillWidth: true
    Component {
        id: tabButton
        TabButton {
            width: implicitWidth
        }
    }
    Component {
        id: resultListView
        ResultList {}
    }
    SwipeView {
        id: tabBarLayout
        Layout.fillHeight: true
        Layout.fillWidth: true
        interactive: false
        currentIndex: tabBar.currentIndex
        LogArea {
            id: logArea
        }
    }
    TabBar {
        id: tabBar
        currentIndex: tabBarLayout.currentIndex
        Layout.fillWidth: true
        font.capitalization: Font.MixedCase
        TabButton {
            text: qsTr("log")
            width: implicitWidth
            onClicked: tabBar.setCurrentIndex(0)
        }

        Connections {
            target: qmlBridge
            onLoadResultTab: {
                tabBar.addItem(tabButton.createObject(tabBar, {text: qsTr("search - " + term)}))
                tabBar.incrementCurrentIndex()
                tabBarLayout.addItem(resultListView.createObject())
                console.log("Added new item...")
            }
        }
    }
}

```

The `onLoadResultTab` signal is invoked when the button is clicked. As you can see, I'm adding a new `TabButton` and incrementing the current index and then creating another component which is called the `ResultList`. Now, when I click on the "log" button, the `SwipeView` doesn't change its page to the one corresponding to the log tab. Could anyone point out what's the issue here?

**Comment by therecipe** [comment]
Hey

I'm no Qml expert by any means, but it could be that your `onLoadResultTab` function is wrong.
Maybe try to add both items and then increment the current index.

Also, I couldn't directly reproduce the issue but this example seems to work for me:
```qml
import QtQuick 2.12
import QtQuick.Layouts 1.3
import QtGraphicalEffects 1.12
import QtQuick.Controls 2.13

ColumnLayout {
    anchors.fill: parent

    Component {
        id: tabButton
        TabButton {
            width: implicitWidth
        }
    }

    Component {
        id: resultListView
        Rectangle {
          width: tabBarLayout.width
          height: tabBarLayout.height
        }
    }

    SwipeView {
        id: tabBarLayout

        Layout.fillHeight: true
        Layout.fillWidth: true

        interactive: false
        currentIndex: tabBar.currentIndex

        Rectangle {
          width: tabBarLayout.width
          height: tabBarLayout.height
          color: "blue"
        }
    }

    TabBar {
        id: tabBar
        currentIndex: tabBarLayout.currentIndex
        Layout.fillWidth: true
        TabButton {
            text: qsTr("log")
            width: implicitWidth
        }
    }

    Button {
    	onClicked: {
                tabBar.addItem(tabButton.createObject(tabBar, {text: qsTr("search - " + (tabBar.count))}));
               
                var col = "green";
                if (tabBar.currentIndex % 2 == 0){
                	col = "red";
                }
                tabBarLayout.addItem(resultListView.createObject(tabBarLayout, {color: col}))

                tabBar.setCurrentIndex(tabBar.count - 1);
                console.log("Added new item...")
    	}
    }
}
```

(I also made some changes to the `onLoadResultTab` function to use `tabBar.count` instead, but it did seem to work with `tabBar.incrementCurrentIndex()` as well.)

**Comment by gravetii** [comment]
Hey @therecipe , thank you so much for taking time out to look at this issue, especially because it's not an issue related to the bindings.

Actually, I looked at your code, and the issue seems to be the layout settings for the `ResultList`. Using the following worked for me, as you described in your solution - 

```
Rectangle {
          width: tabBarLayout.width
          height: tabBarLayout.height
        }
```
I was instead setting the `Layout.fillHeight` & `Layout.fillWidth` properties to true. Seems like I need to understand them much better. I'll do that, and thanks again for spending time on this issue. Closing it now! :)
